native/cocos/core/Root.cpp

```diff
@@ -69,7 +73,40 @@ Root::~Root() {
 }
 
 void Root::initialize(gfx::Swapchain *swapchain) {
-    _swapchain = swapchain;
+    //_swapchain = swapchain;
+
+    ISystemWindowManager *windowMgr = CC_GET_PLATFORM_INTERFACE(ISystemWindowManager);
+    const auto &windows = windowMgr->getWindows();
+    for (auto pair : windows) {
+        auto *window = pair.second.get();
+        _mainWindow = createRenderWindowFromSystemWindow(window);
```

Comment:
`_mainWindow ` will be initialized many times, is it correct?

Reply:
`_mainWindow  `needs to be removed in the future.
How about
 `if (!_mainWindow )
     _mainWindow = createRenderWindowFromSystemWindow(window);`  for now ? @minggo

Reply:
Should define what's the main window. Can make sure the first one created is the main window?

Reply:
Maybe `_mainWindow`  can be specified using setter?

Reply:
Can we rename it to `_mainRenderWindow`?

Reply:
We should distinguish system window and render window, and not just use window. window is confused, don't know it is a render window or a system window.